SMS/SMSMessages.cpp

```diff
@@ -173,11 +173,15 @@ TLMessage *SMS::parseTPDU(const TLFrame& TPDU)
 	LOG(DEBUG) << "SMS: parseTPDU MTI=" << TPDU.MTI();
 	// Handle just the uplink cases.
 	switch ((TLMessage::MessageType)TPDU.MTI()) {
-		case TLMessage::DELIVER_REPORT:
 		case TLMessage::STATUS_REPORT:
 			// FIXME -- Not implemented yet.
 			LOG(WARNING) << "Unsupported TPDU type: " << (TLMessage::MessageType)TPDU.MTI();
 			return NULL;
+		case TLMessage::DELIVER_REPORT: {
```

Comment:
I hope that the parseBody helpers ought to be working as they should. I am not sure any deliver messages will get triggered in a live setting, but it would be nice to test this out before putting it into production